homeassistant/components/filter/config_flow.py

```diff
@@ -0,0 +1,211 @@
+            ),
+            vol.Optional(
+                CONF_FILTER_RADIUS, default=DEFAULT_FILTER_RADIUS
+            ): NumberSelector(
+                NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+            ),
+        }
+
+    elif handler.options[CONF_FILTER_NAME] == FILTER_NAME_LOWPASS:
+        filter_schema = {
+            vol.Optional(
+                CONF_FILTER_WINDOW_SIZE, default=DEFAULT_WINDOW_SIZE
+            ): NumberSelector(
+                NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+            ),
+            vol.Optional(
+                CONF_FILTER_TIME_CONSTANT, default=DEFAULT_FILTER_TIME_CONSTANT
+            ): NumberSelector(
+                NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+            ),
+        }
+
+    elif handler.options[CONF_FILTER_NAME] == FILTER_NAME_RANGE:
+        filter_schema = {
+            vol.Optional(CONF_FILTER_LOWER_BOUND): NumberSelector(
+                NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+            ),
+            vol.Optional(CONF_FILTER_UPPER_BOUND): NumberSelector(
+                NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+            ),
+        }
+
+    elif handler.options[CONF_FILTER_NAME] == FILTER_NAME_TIME_SMA:
+        filter_schema = {
+            vol.Optional(CONF_TIME_SMA_TYPE, default=TIME_SMA_LAST): SelectSelector(
+                SelectSelectorConfig(
+                    options=[TIME_SMA_LAST],
+                    mode=SelectSelectorMode.DROPDOWN,
+                    translation_key=CONF_TIME_SMA_TYPE,
+                )
+            ),
+            vol.Required(CONF_FILTER_WINDOW_SIZE): DurationSelector(
+                DurationSelectorConfig(enable_day=False, allow_negative=False)
+            ),
+        }
+
+    elif handler.options[CONF_FILTER_NAME] == FILTER_NAME_THROTTLE:
+        filter_schema = {
+            vol.Optional(
+                CONF_FILTER_WINDOW_SIZE, default=DEFAULT_WINDOW_SIZE
+            ): NumberSelector(
+                NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+            ),
+        }
+
+    elif handler.options[CONF_FILTER_NAME] == FILTER_NAME_TIME_THROTTLE:
+        filter_schema = {
+            vol.Required(CONF_FILTER_WINDOW_SIZE): DurationSelector(
+                DurationSelectorConfig(enable_day=False, allow_negative=False)
+            ),
+        }
+
+    base_schema: dict[vol.Marker, Selector] = {
+        vol.Optional(CONF_FILTER_PRECISION, default=DEFAULT_PRECISION): NumberSelector(
+            NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+        )
+    }
+
+    return vol.Schema({**filter_schema, **base_schema})
+
+
+async def validate_options(
+    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate options selected."""
+
+    if CONF_FILTER_WINDOW_SIZE in user_input and isinstance(
```

Comment:
when is it float ?

Reply:
NumberSelector always returns a float so therefore it's converted to an int here